starfish/util/overlap_utils.py

```diff
@@ -0,0 +1,162 @@
+                        max(area1.min_y, area2.min_y),
+                        min(area1.max_y, area2.max_y))
+        return None
+
+
+def find_overlaps_of_xarrays(xarrays: List[xr.DataArray]
+                             ) -> Dict[Tuple[int, int], "Area"]:
+    """
+    Find all the overlap areas within a list of xarrays.
+
+    Parameters
+    ----------
+    xarrays : List[xr.DataArray]
+        The list of xarrays to find overlaps in.
+
+    Returns
+    -------
+    Dict[Tuple[int, int], "Area"]] :
+        A dictionary of tuples containing the indices of two overlapping
+        IntensityTables and their Area of intersection.
+
+    """
+    all_overlaps: Dict[Tuple[int, int], "Area"] = dict()
+    for idx1, idx2 in itertools.combinations(range(len(xarrays)), 2):
+        xr1 = xarrays[idx1]
+        xr2 = xarrays[idx2]
+        area1 = Area(min(xr1[Coordinates.X.value]).data,
+                     max(xr1[Coordinates.X.value]).data,
+                     min(xr1[Coordinates.Y.value]).data,
+                     max(xr1[Coordinates.Y.value]).data)
+
+        area2 = Area(min(xr2[Coordinates.X.value]).data,
+                     max(xr2[Coordinates.X.value]).data,
+                     min(xr2[Coordinates.Y.value]).data,
+                     max(xr2[Coordinates.Y.value]).data)
+        intersection = Area.find_intersection(area1, area2)
+        if intersection:
+            all_overlaps[(idx1, idx2)] = intersection
+    return all_overlaps
+
+
+def remove_area_of_xarray(it: xr.DataArray, area: Area) -> xr.DataArray:
+    """Return everything in the xarray defined OUTSIDE the input area
+
+    Parameters
+    ----------
+    it: xr.DataArray
+        The xarray to modify
+    area: Area
+        The area to not include in the modified xarray
+
+    Returns
+    -------
+     xr.DataArray :
+        The xarray without the defined area.
+    """
+    return it.where((it.xc <= area.min_x)
+                    | (it.xc >= area.max_x)
+                    | (it.yc <= area.min_y)
+                    | (it.yc >= area.max_y),
+                    drop=True)
+
+
+def sel_area_of_xarray(it: xr.DataArray, area: Area) -> xr.DataArray:
+    """Return everything in the xarray defined WITHIN the input area
+
+    Parameters
+    ----------
+    it: xr.DataArray
+        The xarray to modify
+    area: Area
+        The area to include in the modified xarray
+
+    Returns
+    -------
+     xr.DataArray :
+        The xarray within the defined area.
+    """
+    return it.where((it.xc >= area.min_x)
+                    & (it.xc <= area.max_x)
+                    & (it.yc >= area.min_y)
+                    & (it.yc <= area.max_y), drop=True)
+
+
+def take_max(intersection_rect: Area,
+             it1: xr.DataArray,
+             it2: xr.DataArray
+             ):
+    """
+    Compare two overlapping xarrays and remove spots from whichever
+    has less in the overlapping section.
+
+    Parameters
+    ----------
+    intersection_rect : Area
```

Comment:
Why not calculate this internally?

Reply:
I guess I just didn't want to calculate it twice since we'd already done it to find the overlap pairs. But you're right it might be better for find_overlaps_of_xarrays() to just output a list of overlapping pairs and for take_max to actually find the overlap.

Reply:
This does make my tests a little less useful since I was validating that find_overlaps_of_xarrays() finds the right overlap pairs AND the right overlapping area, and now it would just verify it finds the overlapping pairs. What do you think?